tests/setup_test.go

```diff
@@ -3130,3 +3134,442 @@ func TestGetUsers(t *testing.T) {
+				}
+				if err.Error() != expectedErr {
+					t.Fatalf("expected err %v, got err: %v", expectedErr, err)
+				}
+			},
+		},
+		{
+			name: "test admin user enable maintenance just enabled",
+			f: func(ctx context.Context, t *testing.T, sc *setupContext) {
+				gwClient := gwclient.NewClient(sc.config.Gateway.APIExposedURL, sc.config.Gateway.AdminToken)
+
+				_, err := gwClient.EnableMaintenance(ctx, configstoreService)
+				if err != nil {
+					t.Fatalf("unexpected err: %v", err)
+				}
+				if err != nil {
+					t.Fatalf("unexpected err: %v", err)
+				}
+
+				_, err = gwClient.EnableMaintenance(ctx, runserviceService)
+				if err != nil {
+					t.Fatalf("unexpected err: %v", err)
+				}
+
+				expectedErr := "remote error badrequest"
+				_, err = gwClient.EnableMaintenance(ctx, configstoreService)
+				if err == nil {
+					t.Fatalf("expected error %v, got nil err", expectedErr)
+				}
+				if err.Error() != expectedErr {
+					t.Fatalf("expected err %v, got err: %v", expectedErr, err)
+				}
+
+				_, err = gwClient.EnableMaintenance(ctx, runserviceService)
+				if err == nil {
+					t.Fatalf("expected error %v, got nil err", expectedErr)
+				}
+				if err.Error() != expectedErr {
+					t.Fatalf("expected err %v, got err: %v", expectedErr, err)
+				}
+			},
+		},
+		{
+			name: "test admin user disable maintenance",
+			f: func(ctx context.Context, t *testing.T, sc *setupContext) {
+				gwClient := gwclient.NewClient(sc.config.Gateway.APIExposedURL, sc.config.Gateway.AdminToken)
+
+				_, err := gwClient.EnableMaintenance(ctx, configstoreService)
+				if err != nil {
+					t.Fatalf("unexpected err: %v", err)
+				}
+
+				_, err = gwClient.DisableMaintenance(ctx, configstoreService)
+				if err != nil {
+					t.Fatalf("unexpected err: %v", err)
+				}
+
+				_, err = gwClient.EnableMaintenance(ctx, runserviceService)
+				if err != nil {
+					t.Fatalf("unexpected err: %v", err)
+				}
+
+				_, err = gwClient.DisableMaintenance(ctx, runserviceService)
+				if err != nil {
+					t.Fatalf("unexpected err: %v", err)
+				}
+			},
+		},
+		{
+			name: "test admin user disable maintenance just disabled",
+			f: func(ctx context.Context, t *testing.T, sc *setupContext) {
+				gwClient := gwclient.NewClient(sc.config.Gateway.APIExposedURL, sc.config.Gateway.AdminToken)
+
+				expectedErr := "remote error badrequest"
+				_, err := gwClient.DisableMaintenance(ctx, configstoreService)
+				if err == nil {
+					t.Fatalf("expected error %v, got nil err", expectedErr)
+				}
+				if err.Error() != expectedErr {
+					t.Fatalf("expected err %v, got err: %v", expectedErr, err)
+				}
+
+				_, err = gwClient.DisableMaintenance(ctx, runserviceService)
+				if err == nil {
+					t.Fatalf("expected error %v, got nil err", expectedErr)
+				}
+				if err.Error() != expectedErr {
+					t.Fatalf("expected err %v, got err: %v", expectedErr, err)
+				}
+			},
+		},
+		{
+			name: "test admin user maintenance badrequest",
```

Comment:
is this meant to test a wrong provided servicename? Is so the test should be named correctly (not badrequest)